config/eventhub/config.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventhub
+
+import (
+	"github.com/crossplane/terrajet/pkg/config"
+
+	"github.com/crossplane-contrib/provider-jet-azure/config/common"
+)
+
+// Configure configures resource group
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("azurerm_eventhub_namespace", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.Kind = "EventHubNamespace"
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Eventhub/namespaces/namespace1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.Eventhub",
+			"namespaces", "name",
+		)
+		r.LateInitializer = config.LateInitializer{
+			IgnoredFields: []string{"network_rulesets"},
+		}
+	})
+
+	p.AddResourceConfigurator("azurerm_eventhub", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.Kind = "EventHub"
+		r.References = config.References{
+			"namespace_name": config.Reference{
+				Type: "EventHubNamespace",
+			},
+		}
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.EventHub/namespaces/namespace1/eventhubs/eventhub1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.Eventhub",
+			"namespaces", "name",
+			"eventhubs", "name",
+		)
+	})
+
+	p.AddResourceConfigurator("azurerm_eventhub_consumer_group", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.Kind = "ConsumerGroup"
+		r.References = config.References{
+			"namespace_name": config.Reference{
+				Type: "EventHubNamespace",
+			},
+			"eventhub_name": config.Reference{
+				Type: "EventHub",
+			},
+		}
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.EventHub/namespaces/namespace1/eventhubs/eventhub1/consumerGroups/consumerGroup1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.Eventhub",
+			"namespaces", "name",
+			"eventhubs", "name",
+			"consumergroups", "name",
```

Comment:
```suggestion
			"consumerGroups", "name",
```
Looks like it's working as it is. But, I'd change it anyways.

Reply:
@ezgidemirel good catch, but surprisingly it fails with capital `consumerGroups`. I did it initially according to the docs and it failed to find ID. Looks like a bug on azure side. It is actually inconsistent behavior if to compare with https://github.com/crossplane-contrib/provider-jet-azure/pull/149/files#diff-1900d6efdae13993ebd24b09af5b96aa2ca693f16dad96fef4b0c6295a2c8946R100.